app/models/concerns/has_fields.rb

```diff
@@ -56,10 +56,19 @@ def dradis_has_fields_for(container_field)
         updated_fields[field] = value
         self.send(
           :"#{container_field}=",
-          updated_fields.to_a.map { |h| "#[#{h[0]}]#\n#{h[1]}" }.join("\n\n")
+          FieldParser.fields_hash_to_source(updated_fields)
+        )
+      end
+
+      # Completely removes the field (field header and value) from the content
+      define_method :delete_field do |field|
+        updated_fields = fields
+        updated_fields.except!(field)
+        self.send(
+          :"#{container_field}=",
+          FieldParser.fields_hash_to_source(updated_fields)
```

Comment:
this is the same as L57-L70, is it worth extracting into a (private?) helper (e.g. #save_fields - mave "save" isn't right; #update_container)?